src/watch.rs

```diff
@@ -43,22 +52,27 @@ impl Watch {
     }
 
     /// Wait for a batch of changes to arrive, returning when they do.
-    pub fn wait_for_change(&mut self) -> Result<(), ()> {
+    pub fn wait_for_change(&mut self) -> Result<Reason, ()> {
         self.block()
     }
 
     /// Block until we have at least one event
-    pub fn block(&mut self) -> Result<(), ()> {
-        if self.blocking_iter().next().is_none() {
-            debug!("No event received!");
-            return Err(());
+    pub fn block(&mut self) -> Result<Reason, ()> {
+        match self.blocking_iter().next() {
+            Some(event) => {
+                let extra = self.process_ready();
+                Ok(Reason::FilesChanged(event.path.ok_or(())?, extra?))
```

Comment:
I noticed that `process_ready` does not actually return an `Err`, so you can change that to a plain `u16` and remove the `?` on `extra`.